tool/tctl/common/auth_command_test.go

```diff
@@ -633,3 +639,112 @@ func TestGenerateAppCertificates(t *testing.T) {
 		})
 	}
 }
+
+func TestGenerateDatabaseUserCertificates(t *testing.T) {
+	tests := map[string]struct {
+		clusterName        string
+		dbName             string
+		dbUser             string
+		expectedDbProtocol string
+		dbServices         []types.DatabaseServer
+		withError          bool
+	}{
+		"DatabaseExists": {
+			clusterName:        "example.com",
+			dbName:             "db-1",
+			expectedDbProtocol: defaults.ProtocolPostgres,
+			dbServices: []types.DatabaseServer{
+				&types.DatabaseServerV3{
+					Metadata: types.Metadata{
+						Name: "db-1",
+					},
+					Spec: types.DatabaseServerSpecV3{
+						Hostname: "example.com",
+						Database: &types.DatabaseV3{
+							Spec: types.DatabaseSpecV3{
+								Protocol: defaults.ProtocolPostgres,
+							},
+						},
+					},
+				},
+			},
+		},
+		"DatabaseWithUserExists": {
+			clusterName:        "example.com",
+			dbName:             "db-user-1",
+			dbUser:             "mongo-user",
+			expectedDbProtocol: defaults.ProtocolMongoDB,
+			dbServices: []types.DatabaseServer{
+				&types.DatabaseServerV3{
+					Metadata: types.Metadata{
+						Name: "db-user-1",
+					},
+					Spec: types.DatabaseServerSpecV3{
+						Hostname: "example.com",
+						Database: &types.DatabaseV3{
+							Spec: types.DatabaseSpecV3{
+								Protocol: defaults.ProtocolMongoDB,
+							},
+						},
+					},
+				},
+			},
+		},
+		"DatabaseNotFound": {
+			clusterName: "example.com",
+			dbName:      "db-2",
+			dbServices:  []types.DatabaseServer{},
+			withError:   true,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			clusterName, err := services.NewClusterNameWithRandomID(
+				types.ClusterNameSpecV2{
+					ClusterName: test.clusterName,
+				})
+			require.NoError(t, err)
+
+			authClient := &mockClient{
+				clusterName: clusterName,
+				userCerts: &proto.Certs{
+					SSH: []byte("SSH cert"),
+					TLS: []byte("TLS cert"),
+				},
+				dbServices: test.dbServices,
+			}
+
+			certsDir := t.TempDir()
+			output := filepath.Join(certsDir, test.dbName)
+			ac := AuthCommand{
+				output:        output,
+				outputFormat:  identityfile.FormatTLS,
+				signOverwrite: true,
+				genTTL:        time.Hour,
+				dbName:        test.dbName,
+				dbUser:        test.dbUser,
+			}
+
+			err = ac.generateUserKeys(authClient)
+			if test.withError {
+				require.Error(t, err)
+				return
+			}
+
+			require.NoError(t, err)
+
+			expectedRouteToDatabase := proto.RouteToDatabase{
+				ServiceName: test.dbName,
+				Protocol:    test.expectedDbProtocol,
+				Username:    test.dbUser,
+			}
+			require.Equal(t, proto.UserCertsRequest_Database, authClient.userCertsReq.Usage)
+			require.Equal(t, expectedRouteToDatabase, authClient.userCertsReq.RouteToDatabase)
+
+			certBytes, err := ioutil.ReadFile(filepath.Join(certsDir, test.dbName+".crt"))
```

Comment:
nit: `ioutils` is deprecated
```suggestion
			certBytes, err := os.ReadFile(filepath.Join(certsDir, test.dbName+".crt"))
```

Reply:
Updated the entire file not to use this package anymore.